src/custom/components/Header/MenuTree/index.tsx

```diff
@@ -0,0 +1,165 @@
+      </ExternalLinkComponent>
+    )
+  } else {
+    return (
+      <StyledNavLink exact to={url} onClick={handleMobileMenuOnClick}>
+        {menuImage}
+        {title}
+      </StyledNavLink>
+    )
+  }
+}
+
+interface ContextProps {
+  darkMode: boolean
+  toggleDarkMode: () => void
+  handleMobileMenuOnClick: () => void
+}
+
+type DarkModeButtonProps = {
+  context: ContextProps
+}
+
+function DarkModeButton({ context }: DarkModeButtonProps) {
+  const { darkMode, toggleDarkMode, handleMobileMenuOnClick } = context
+  const description = `${darkMode ? 'Sun/light' : 'Moon/dark'} mode icon`
+  const label = (darkMode ? 'Light' : 'Dark') + ' Mode'
+  return (
+    <button
+      onClick={() => {
+        handleMobileMenuOnClick()
+        toggleDarkMode()
+      }}
+    >
+      <SVG src={darkMode ? IMAGE_SUN : IMAGE_MOON} description={description} /> {label}
+    </button>
+  )
+}
+
+interface LinkProps {
+  link: MenuLink
+  context: ContextProps
+}
+
+function Link({ link, context }: LinkProps) {
+  switch (link.kind) {
+    case MenuItemKind.DARK_MODE_BUTTON:
+      return <DarkModeButton context={context} />
+
+    default:
+      return <InternalExternalLink link={link} handleMobileMenuOnClick={context.handleMobileMenuOnClick} />
+  }
+}
+
+interface DropdownProps {
+  item: DropDownItem
+  context: ContextProps
+}
+
+const DropDown = ({ item, context }: DropdownProps) => {
+  const { title, items } = item
+
+  return (
+    <MenuDropdown title={title}>
+      {items?.map((item, index) => {
+        const { sectionTitle, links } = item
+        return (
+          <MenuSection key={index}>
+            {sectionTitle && <MenuTitle>{sectionTitle}</MenuTitle>}
+            {links.map((link, linkIndex) => (
+              <Link key={linkIndex} link={link} context={context} />
+            ))}
+          </MenuSection>
+        )
+      })}
+    </MenuDropdown>
+  )
+}
+
+interface MenuItemWithDropDownProps {
+  menuItem: MenuTreeItem
+  context: ContextProps
+}
+
+function MenuItemWithDropDown(props: MenuItemWithDropDownProps) {
+  const { menuItem, context } = props
+
+  switch (menuItem.kind) {
+    case MenuItemKind.DROP_DOWN:
+      return <DropDown item={menuItem} context={context} />
+
+    case undefined: // INTERNAL
+    case MenuItemKind.EXTERNAL_LINK: // EXTERNAL
+      // Render Internal/External links
+      return <InternalExternalLink link={menuItem} handleMobileMenuOnClick={context.handleMobileMenuOnClick} />
+    default:
+      return null
+  }
+}
+
+export interface MenuTreeProps extends ContextProps {
+  isMobileMenuOpen: boolean
+}
+
+export function MenuTree({ isMobileMenuOpen, darkMode, toggleDarkMode, handleMobileMenuOnClick }: MenuTreeProps) {
+  const context = { darkMode, toggleDarkMode, handleMobileMenuOnClick }
+  return (
+    <Wrapper isMobileMenuOpen={isMobileMenuOpen}>
+      {MAIN_MENU.map((menuItem, index) => (
+        <MenuItemWithDropDown key={index} menuItem={menuItem} context={context} />
```

Comment:
As David mentioned we could change this here also, but since these links are hard-coded and not the data tha is changing I think using index'es as keys is also fine.

Reply:
I think is not worth it.

See:
https://github.com/cowprotocol/cowswap/pull/715#discussion_r902468677